src/engine/openmp.cc

```diff
@@ -83,10 +83,10 @@ void OpenMP::set_reserve_cores(int cores) {
 
 int OpenMP::GetRecommendedOMPThreadCount(bool exclude_reserved) const {
 #ifdef _OPENMP
-  if (omp_num_threads_set_in_environment_) {
-    return omp_get_max_threads();
-  }
   if (enabled_) {
+    if (omp_num_threads_set_in_environment_) {
```

Comment:
Can we add some comments on under which scenario this would happen? The logic in this class has become too complicated to follow.

Reply:
sure, makes sense.

Reply:
Is it better now?